src/Microsoft.Extensions.Configuration.AzureAppConfiguration/OfflineFileCache.cs

```diff
@@ -144,23 +111,29 @@ public string Import(AzureAppConfigurationOptions options)
                         }
                     }
 
-                    if ((data != null) && (dataHash != null) && (scopeHash != null))
+                    if ((data != null) && (scope != null) && (expiryString != null) && (_scopeToken == scope))
                     {
-                        string newScopeHash = CryptoService.GetHash(Encoding.UTF8.GetBytes(_scopeToken), _options.SignKey);
-                        if (string.CompareOrdinal(scopeHash, newScopeHash) == 0)
+                        if (DateTimeOffset.TryParse(expiryString, CultureInfo.InvariantCulture, DateTimeStyles.None, out DateTimeOffset expiryTime) 
+                            && DateTimeOffset.UtcNow < expiryTime)
                         {
-                            string newDataHash = CryptoService.GetHash(Convert.FromBase64String(data), _options.SignKey);
-                            if (string.CompareOrdinal(dataHash, newDataHash) == 0)
-                            {
-                                return CryptoService.AESDecrypt(data, _options.Key, _options.IV);
-                            }
+                            return _timeLimitedDataProtector.Unprotect(data);
                         }
                     }
                 }
                 catch (IOException ex) when (ex.HResult == ERROR_SHARING_VIOLATION)
                 {
                     Task.Delay(new Random().Next(delayRange)).ConfigureAwait(false).GetAwaiter().GetResult();
                 }
+                catch (CryptographicException)
+                {
+                    // TBD: This should be handled in AzureAppConfigurationProvider class.
+                    // This exception means that one of the following occured:
+                    //  - the encryption key is invalid/inaccessible;
+                    //  - encrypted data has been tampered with;
+                    //  - data has expired;
+                    //  - any other internal error during decryption.
+                    throw;
+                }

```

Comment:
This doesn't look like it's being used for anything. Should be able to remove the catch

Reply:
I wanted to keep it as a to-do here so that we remember to catch it in `AzureAppConfigurationProvider` as required. In the final version, we shouldn't have to catch it here.